lib/src/api/engine/local/native.rs

```diff
@@ -141,18 +147,45 @@ pub(crate) fn router(
 			false => kvs,
 		};
 
+		let kvs = Arc::new(kvs);
 		let mut vars = BTreeMap::new();
-		let mut stream = route_rx.into_stream();
 		let mut session = Session::default();
 
-		while let Some(Some(route)) = stream.next().await {
-			match super::router(route.request, &kvs, &mut session, &mut vars).await {
-				Ok(value) => {
-					let _ = route.response.into_send_async(Ok(value)).await;
+		let tick_interval = address.config.tick_interval.unwrap_or(DEFAULT_TICK_INTERVAL);
+		let mut interval = time::interval(tick_interval);
+		// Don't bombard the database if we miss some ticks
+		interval.set_missed_tick_behavior(MissedTickBehavior::Delay);
+		// Delay sending the first tick
+		interval.tick().await;
+
+		let ticker = IntervalStream::new(interval);
+
+		let streams = (route_rx.into_stream().map(Either::Left), ticker.map(|_| Either::Right(())));
+
+		let mut stream = streams.merge();
+
+		while let Some(either) = stream.next().await {
+			match either {
+				Either::Left(Some(route)) => {
+					match super::router(route.request, &kvs, &mut session, &mut vars).await {
+						Ok(value) => {
+							let _ = route.response.into_send_async(Ok(value)).await;
+						}
+						Err(error) => {
+							let _ = route.response.into_send_async(Err(error)).await;
+						}
+					}
 				}
-				Err(error) => {
-					let _ = route.response.into_send_async(Err(error)).await;
+				Either::Right(()) => {
+					let kvs = kvs.clone();
+					tokio::spawn(async move {
+						match kvs.tick().await {
+							Ok(()) => trace!("Node agent tick ran successfully"),
+							Err(error) => error!("Error running node agent tick: {error}"),
+						}
+					});
```

Comment:
This might produce two or more concurrently running kvs.tick() when it takes long time, in case there are a lot of ns/db in the database!

Quick idea; Can we instead of a long running tokio::spawn async function outside of this stream handler? Perhaps do another tokio::spawn in L97 do ticking there? 🤔

Reply:
Good point. I wasn't aware that `tick` can run long enough to potentially overlap with others and that such overlap would be bad. With the current implementation cancellation comes for free, that's why I implemented it like that. I will change it to address your concern 👍